### ISSUE TITLE
transformer: incorrect semantic check

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#4999]
I saw nearly all tests have a `Symbols mismatch after transform` error. This error is thrown here

https://github.com/oxc-project/oxc/blob/081e2a37d96a765216f9bb0025208a9fdd699667/crates/oxc_semantic/src/post_transform_checker.rs#L129-L132

After looking into it, what happened here, I found a bug in `PostTransformChecker`. From the code above, it can be seen that this is due to the unequal length of semantic symbols in the two different phases. 

The `previous_collect` is collected in `after_semantic` and the `current_collect` is collected in `after_transform`. This is the cause of the problem. After the transformation, the `Semantic` will definitely change. 

So we shouldn't use `after_semantic`'s semantic to compare with `after_transform`'s semantic. Instead, we should rebuild the semantic after the transformation, and compare it to `after_transform`'s semantic

**Comment by Dunqing** [comment]
We may need to solve this problem as soon as possible. Since we can't catch valid any regression in semantic changes, I am a bit afraid to refactor the transformer 🥲

**Comment by overlookmotel** [comment]
I'm going to look at this tomorrow. Will try to wrap it up as quickly as possible.

**Comment by Boshen** [comment]
@Dunqing Can you give us a test case? Ideally what's wrong and the expected result?

**Comment by Dunqing** [comment]
In the following test used `nullish-coalescing-operators` plugin

#### Input
```js
// nullish-coalescing/transform-in-function/input.js
function foo(opts) {
  var foo = opts.foo ?? "default";
}

```

#### Output
```js
// nullish-coalescing/transform-in-function/output.js
function foo(opts) {
  var _opts$foo;
  var foo = (_opts$foo = opts.foo) !== null && _opts$foo !== void 0 ? _opts$foo : "default";
}
```

The `previous_collect.symbol_ids` have 3 symbols, They are `foo(function name)`, `opts`, and `foo(var foo)`.
The `current_collect.symbol_ids` have 4 symbols, They are `foo(function name)`, `opts`, `_opts$foo`, and `foo(var foo).

Both of the different phases' symbols are correct. The main cause is we use `after_semantic`'s `Semantic` to compare with `after_transform`'s `Semantic`.

**Comment by overlookmotel** [comment]
Fixed in #5035.
